src/main/kotlin/com/exactpro/th2/codec/xml/XmlMessageStructureVisitor.kt

```diff
@@ -45,7 +45,13 @@ class XmlMessageStructureVisitor(private val document: Document, private val nod
                     error("Field node is not element. Field name = $fieldName")
                 }
             } else {
-                node.addNode(fldStruct.getXmlTagName(), document).setText(fieldValue, document)
+                val nodeName = fldStruct.getXmlTagName()
+                val valueNode = node.findNode(nodeName)
+                if (valueNode == null) {
+                    node.addNode(nodeName, document).setText(fieldValue, document)
+                } else if (valueNode.getText() != fieldValue) {
+                    error("Can not encode field with name '$fieldName' with two different values: ['$fieldValue','${valueNode.getText()}']")
+                }
```

Comment:
This check will be executed only when the parent and child virtual message has fields with equals names. But we can prohibit this behavior.